tools/testsys/src/crds.rs

```diff
@@ -159,9 +181,20 @@ pub(crate) trait CrdCreator: Sync {
     /// Create a testing CRD for this variant of Bottlerocket.
     async fn test_crd<'a>(&self, test_input: TestInput<'a>) -> Result<CreateCrdOutput>;
 
+    /// Create a set of additional fields that may be used by an externally defined agent on top of
+    /// the ones in `CrdInput`
+    fn additional_fields(&self, _test_type: &str) -> BTreeMap<String, String> {
+        Default::default()
+    }
+
     /// Creates a set of CRDs for the specified variant and test type that can be added to a TestSys
     /// cluster.
-    async fn create_crds(&self, test_type: TestType, crd_input: &CrdInput) -> Result<Vec<Crd>> {
+    async fn create_crds(
+        &self,
+        test_type: TestType,
+        crd_input: &CrdInput,
+        crd_template_file_path: Option<String>,
```

Comment:
This function signature isn't ideal. You have an Enum, `test_type` and a field `crd_template_file_path` that is required only when `test_type` is `Unknown`.

Can you break up this function into, say, 2 functions, e.g. `create_crds` (which takes `KnownTestType`) and `create_custom_crds` which takes the name of the custom test and the `crd_template_file_path`?

This solves two problems, a quirky function signature, and a function definition that is already long. The caller would have to call either `create_crds` or `create_custom_crds`.

Incidentally, would `Custom` be a better name than `Unknown` throughout?